src/main/java/com/elastic/support/diagnostics/chain/DiagnosticChainExec.java

```diff
@@ -2,22 +2,42 @@
 
 import com.elastic.support.config.Constants;
 import com.elastic.support.diagnostics.DiagnosticException;
+import com.elastic.support.diagnostics.commands.*;
 import com.elastic.support.util.SystemProperties;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import java.util.List;
-import java.util.Map;
 
 public class DiagnosticChainExec {
 
     private static Logger logger = LogManager.getLogger(DiagnosticChainExec.class);
 
-    public void runDiagnostic(DiagnosticContext context, List<String> chain) {
+    public static void runDiagnostic(DiagnosticContext context, String type) {
 
         try {
-            Chain diagnostic = new Chain(chain);
-            diagnostic.execute(context);
+            if("standard".equals(type) || "remote".equals(type)) {
```

Comment:
Seems like this could be a `switch` statement with `"standard"` falling through to `"remote"`? Unless order of the commands is important.

Also, it seems like this is missing an `else` caveat to catch an invalid type making it through.

Reply:
This is just an interim step to change it over from the configuration file. Type options are going to be changing anyway before the release.